### PR TITLE
#6083 update examples in File part of Native API: 'Provenance' section.

### PR DESCRIPTION
**Pull request by pkiraly**
**What this PR does / why we need it**: It improves the Native API documentation by providing full curl examples for a File regarding call in 'Provenance' section
**Which issue(s) this PR closes**: none of them. It is an additional step for solving #6083

**Special notes for your reviewer**: This PR continue previous pull requests which handles examples regarding dataverses/datasets/files (e.g. #6600). In the "Create/Update Provenance JSON and provide related entity name for an uploaded file" it is not clear what is an entity name. The original text doesn't give any entity name, neither Dataverse source code at the API call, so I addedd an extra line:

```
  export ENTITY_NAME="..."
```

Please provide an entity name, and I change it.

**Suggestions on how to test this**:

```
cd doc/sphinx-guides
make clean && make html
firefox build/html/api/native-api.html
```
Check 'Provenance' section

**Does this PR introduce a user interface change?**: no

**Is there a release notes update needed for this change?**: no

**Additional documentation**: no

### CODE REVIEW

**Comment by pkiraly** [comment]
Note: I added subheadings, because otherwise it is hard to follow what example belongs to what topic.

**Comment by djbrooke** [comment]
Thanks @pkiraly. The text looks fine, but I don't know the answer to your entity name question. I'll defer to one of the developers.

**Comment by pkiraly** [comment]
@sekmiller @djbrooke Wait a minute, I had a question, which was not answered: what is entity name? If we leave it as it is, I'll create a new issue asking it again, and when there will be answer, we should update this part.

**Comment by sekmiller** [comment]
Sorry. The entity name is the name that links the provenance file to its data file in Dataverse. It gets saved to the datafile record in dataverse.

**Comment by pkiraly** [comment]
@sekmiller Could you tell me an example? Plus: an example JSON file also would be useful for the reader of these API calls.

**Comment by sekmiller** [comment]
I was just chatting with our Kevin. He had a file that worked but given some code changes it no longer does. The developer who made the changes has left IQSS and I don't have anything at the moment. If you give me a little time I will get a working example for you.

**Comment by pkiraly** [comment]
Sure, take your time, and ping me when you have the info. I'll update the documentation accordingly.

**Comment by pkiraly** [comment]
This PR was developed by Göttingen eResearch Alliance, Germany, and funded by SSHOC, "Social Sciences and Humanities Open Cloud". SSHOC has received funding from the European Union’s Horizon 2020 project call H2020-INFRAEOSC-04-2018, grant agreement #823782.
